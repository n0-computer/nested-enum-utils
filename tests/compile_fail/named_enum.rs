use nested_enum_utils::enum_conversions;

#[derive(Debug)]
#[enum_conversions(Outer)]
enum Enum {
    A { value : u8 },
}

fn main() {}
