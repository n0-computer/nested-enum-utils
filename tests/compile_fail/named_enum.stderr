error: EnumConversions only works with enums that have unnamed single fields
 --> tests/compile_fail/named_enum.rs:6:5
  |
6 |     A { value : u8 },
  |     ^^^^^^^^^^^^^^^^

error[E0412]: cannot find type `Enum` in this scope
 --> tests/compile_fail/named_enum.rs:3:10
  |
3 | #[derive(Debug)]
  |          ^^^^^ not found in this scope
  |
help: consider importing this struct
  |
1 + use syn::token::Enum;
  |

error[E0433]: failed to resolve: use of undeclared type `Enum`
 --> tests/compile_fail/named_enum.rs:6:5
  |
6 |     A { value : u8 },
  |     ^^^^^^^^^^^^^^^^ use of undeclared type `Enum`
  |
help: consider importing this struct
  |
1 + use syn::token::Enum;
  |
